### PR TITLE
Add some real C functions

### PR DESCRIPTION
**Pull request by seanhandley**
Adding various functions with specs.

25 down, 18 to go.

Tracking remainder here https://github.com/StuartApp/h3_ruby/issues/5

@supagroova The repo needs dev permissions sorting out 😁

### CODE REVIEW

**Comment by seanhandley** [comment]
I'd appreciate a review if you don't mind @fxn 😁 

I've done a tiny bit of C before but I'd like some reassurance the code isn't prone to leaks or crashes.

File: ext/h3/h3.c [hunk 1]
```diff
@@ -1,16 +1,173 @@
...
+
+static VALUE h3_stringToH3(VALUE mod, VALUE str) {
+  return LONG2NUM(stringToH3(StringValuePtr(str)));
+}
+
+static VALUE h3_h3ToString(VALUE mod, VALUE h) {
+  size_t sz = 32;
+  char *str = malloc(sz);
+  h3ToString(NUM2LONG(h), str, sz);
+  return rb_str_new_cstr(str);
```
Comment (seanhandley) [inline ext/h3/h3.c]: I guess this'll need a `free()`

**Comment by fxn** [comment]
ACK!

Very cool that you guys wrote this. I want to check a couple of things, will write back after lunch.

**Comment by seanhandley** [comment]
Thanks @fxn - the header file is here: https://github.com/uber/h3/blob/master/src/h3lib/include/h3api.h

File: ext/h3/h3.c [hunk 2]
```diff
@@ -1,16 +1,175 @@
...
+  return LONG2NUM(getH3UnidirectionalEdge(NUM2LONG(origin), NUM2LONG(destination)));
+}
+
+static VALUE h3_stringToH3(VALUE mod, VALUE str) {
+  return LONG2NUM(stringToH3(StringValuePtr(str)));
+}
+
+static VALUE h3_h3ToString(VALUE mod, VALUE h) {
+  size_t sz = 32;
+  char *str = malloc(sz);
```
[the remark this answers is not in the document]
Comment (seanhandley) [reply ext/h3/h3.c]: I'm not sure actually! I think I probably just decided it would be big enough.

[the remark this answers is not in the document]
Comment (seanhandley) [reply ext/h3/h3.c]: Ok, I've changed the code to use this style.

[the remark this answers is not in the document]
Comment (fxn) [reply ext/h3/h3.c]: Regarding the size of the buffer, I have seen that `h3` source code itself allocates arrays of `BUFF_SIZE`, which is initialized to 256 [here](https://github.com/uber/h3/blob/a88fd608361058781cd0f8ec8e4c53149ffdaad4/src/apps/applib/include/utility.h#L26). For example [here](https://github.com/uber/h3/blob/7d36b839cb744dbfc53cd3865478356ef0d6224b/src/apps/filters/h3ToGeo.c#L62-L65), [here](https://github.com/uber/h3/blob/c05ce488f02ecc59b621731d144a07979f2a0518/src/apps/miscapps/h3ToGeoHier.c#L68-L69), and in a few other places.

However, the [documentation](https://github.com/uber/h3/blob/3d529c50d5816ac1fe97008e6559198ed9cb83ce/docs/api/inspection.md#L35) of that function says it should be "at least" of length 17, so 32 should be enough. What do you think about extracting the hard-coded 32s into a macro `#define H3_TO_STR_BUF_SIZE 32` or somesuch?

[the remark this answers is not in the document]
Comment (seanhandley) [reply ext/h3/h3.c]: Sorry - missed this! Follow up PR incoming.

**Pull request merged**